src/proposals/mips/mip-b17/mip-b17.sol

```diff
@@ -0,0 +1,589 @@
+                    /// (10 ** (6 + 8)) * 2 // 6 decimals example
+                    ///    = 2e14
+                    uint256 initialExchangeRate = (10 **
+                        (ERC20(addresses.getAddress(config.tokenAddressName))
+                            .decimals() + 8)) * 2;
+
+                    MErc20Delegator mToken = new MErc20Delegator(
+                        addresses.getAddress(config.tokenAddressName),
+                        ComptrollerInterface(addr.unitroller),
+                        InterestRateModel(addr.irModel),
+                        initialExchangeRate,
+                        config.name,
+                        config.symbol,
+                        mTokenDecimals,
+                        payable(deployer),
+                        addr.mTokenImpl,
+                        ""
+                    );
+
+                    addresses.addAddress(
+                        config.addressesString,
+                        address(mToken),
+                        true
+                    );
+                }
+            }
+        }
+    }
+
+    function afterDeploy(Addresses addresses, address) public override {
+        address governor = addresses.getAddress("TEMPORAL_GOVERNOR");
+        Configs.CTokenConfiguration[]
+            memory cTokenConfigs = getCTokenConfigurations(block.chainid);
+
+        unchecked {
+            for (uint256 i = 0; i < cTokenConfigs.length; i++) {
+                Configs.CTokenConfiguration memory config = cTokenConfigs[i];
+                supplyCaps.push(config.supplyCap);
+                borrowCaps.push(config.borrowCap);
+
+                /// get the mToken
+                mTokens.push(
+                    MToken(addresses.getAddress(config.addressesString))
+                );
+
+                _validateCaps(addresses, config); /// validate supply and borrow caps
+
+                if (
+                    mTokens[i].reserveFactorMantissa() !=
+                    config.reserveFactor &&
+                    mTokens[i].protocolSeizeShareMantissa() != config.seizeShare
+                ) {
+                    mTokens[i]._setReserveFactor(config.reserveFactor);
+                    mTokens[i]._setProtocolSeizeShare(config.seizeShare);
+                    mTokens[i]._setPendingAdmin(payable(governor)); /// set governor as pending admin of the mToken
+                }
+            }
+        }
+    }
+
+    function afterDeploySetup(Addresses addresses) public override {
+        Configs.CTokenConfiguration[]
+            memory cTokenConfigs = getCTokenConfigurations(block.chainid);
+
+        uint256 cTokenConfigsLength = cTokenConfigs.length;
+        unchecked {
+            for (uint256 i = 0; i < cTokenConfigsLength; i++) {
+                Configs.CTokenConfiguration memory config = cTokenConfigs[i];
+                address tokenAddress = addresses.getAddress(
+                    config.tokenAddressName
+                );
+
+                deal(
```

Comment:
why deal is need here? that means temporalGovenror will need to be funded before executing the proposal?

Reply:
Let me fund temporal governor now

Reply:
Funded: https://basescan.org/tx/0x8fa917befe65ecd5380b593344f537b7732954050da6fbed8289978faf754642